README.md

```diff
@@ -54,5 +54,31 @@ will begin to drift slightly from the original design. This is expected and
 there is no expectation that the original design proposal needs to be updated
 to reflect these differences.
 
-The code and our documentation are the ultimate sources of truth. CFPs are merely
-the starting point for the implementation.
+The code and our documentation are the ultimate sources of truth. CFPs are
+the starting point for the implementation, but can also be updated as needed.
+
+# Status
+
+The status of a CFP indicates its maturity. There are four different statuses
+that a CFP can have:
+
+1. Draft: A CFP in any form (Google doc, Github issue, ect.) that has not yet
+been merged into this repository.
+
+2. Implementable: CFPs that have been approved by one committer and merged into
+this repository are given the implementable status. This means that the ideas in the CFP
+have been agreed in principle and the coding work can now be started. When merged,
+the CFP should be up to date and the relevant stakeholders should be in alignment
+even if they are still going through the practical ramifications of how to implement it.
+
+3. Released [Project] X.Y: Everything listed in the CFP as part of the proposal (not future milestones) is merged into the project repository with the listed release.
+
+4. Dormant: This status is given to a CFP that has not been
+implemented for a variety of reasons, like not enough engineering cycles or not
+currently a focus of the project. Any CFP where there is no active effort to
+build the solution can have the dormant status. This serves as a way to preserve
+previous discussions on a solution, but where the solution was not yet agreed
+upon as implementable. Dormant CFPs can be reactivated
+at any time if there is interest in the project. The next step for a dormant CFP is to amend it to be Implementable.
+
+5. Declined: This proposal was considered by the community but ultimately rejected.
```

Comment:
Suggest add:

"The community may come back to the proposal in the future but a new CFP should be used. Rejected proposals are useful as documentation on why the given proposal did not make sense."

Reply:
> why the given proposal did not make sense.

How do we document this part?

If there's a PR and it's closed as not planned, I think the UX should be fairly clear. On the other hand, if the PR is merged into the repository, then the PR itself needs to be updated to encode this information. One of the things we've struggled with a bit as we start to establish the CFP process is that for the most part, developers don't want to document what they're designing. If you then decline their proposal and ask them to put additional effort into documenting why, then they're probably not all that motivated to comply. On the other hand, I guess as committers/maintainers we could take on that work, but in that case we also (a) need buy-in from those folks and (b) could benefit from an example workflow that describes what those folks should do in order to document the decision and why we feel it's important (helps with buy-in).

Reply:
I've updated the wording to "can be" and leaving the implementation that Joe is discussing above to be worked out when we come to that bridge.

Reply:
I think it's worth noting that the default behaviour if we don't resolve that question will be either (a) to leave PRs that have been declined open, or to (b) just close them without merging. Either of these options takes minimal to no effort vs. figuring out the standard we expect in order to merge a declined CFP.

That said, I'm OK with this outcome and we can always improve later, I just want to set expectations that even if we agree in principle that merging declined CFPs is OK, unless we have a specific pattern to follow, we will *not* merge declined PRs.